msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/IBroker.java

```diff
@@ -67,11 +67,17 @@ default IAuthenticationResult parseBrokerAuthResult(String authority, String idT
             if (idToken != null) {
                 builder.idToken(idToken);
                 if (accountId != null) {
-                    String idTokenJson =
-                            JWTParser.parse(idToken).getParsedParts()[1].decodeToString();
+                    String idTokenJson;
+
+                    try {
+                        idTokenJson = new String(Base64.getDecoder().decode(idToken.split("\\.")[1]), StandardCharsets.UTF_8);
```

Comment:
This is a very strange operation (extracting only the body). Why are you doing this?

Reply:
This was copied from similar code in TokenRequestExecutor, which had been parsing the body since that file was first added: https://github.com/AzureAD/microsoft-authentication-library-for-java/blame/bbdc93d59d54916cdbac4016f1b174822e9bb51c/msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/TokenRequestExecutor.java#L123

An IdToken is created from the JSON of claims in the body: https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/dev/msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/IdToken.java

I'm not sure exactly why just the body is decoded, however from some testing now it seems like Java's built-in base64 URL decoded cannot parse the signature so that may be why.